packages/flutter/test/cupertino/popup_surface_test.dart

```diff
@@ -0,0 +1,328 @@
+                  top: a * tileHeight + s * tileHeight / 4,
+                  height: tileHeight,
+                  width: tileWidth,
+                  child: ColoredBox(
+                    color: HSVColor.fromAHSV(
+                      0.5 + a / 8,
+                      h * 45,
+                      0.5 + s / 8,
+                      0.5 + b / 8,
+                    ).toColor(),
+                  ),
+                ),
+        Padding(
+          padding: const EdgeInsets.all(32),
+          child: CupertinoTheme(
+            data: CupertinoThemeData(brightness: brightness),
+            child: _child,
+          ),
+        ),
+      ]),
+    );
+  }
+}
+
+void main() {
+  group('Color filter', () {
+    testWidgets('Brightness.light color filter', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            blurSigma: 0,
+            isSurfacePainted: false,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // Golden displays the color filter effect of the CupertinoPopupSurface
+      // when the ambient brightness is light.
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.color-filter.light.png'),
+      );
+    });
+    testWidgets('Brightness.dark color filter', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            blurSigma: 0,
+            isSurfacePainted: false,
+            child: SizedBox(),
+          ),
+          brightness: Brightness.dark,
+        ),
+      );
+
+      // Golden displays the color filter effect of the CupertinoPopupSurface
+      // when the ambient brightness is dark.
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.color-filter.dark.png'),
+      );
+    });
+    testWidgets('Setting isVibrancePainted to false removes the color filter', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            blurSigma: 0,
+            isSurfacePainted: false,
+            isVibrancePainted: false,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // The BackdropFilter widget should not be mounted when blurSigma is 0 and
+      // CupertinoPopupSurface.isVibrancePainted is false.
+      expect(
+        find.descendant(
+          of: find.byType(CupertinoPopupSurface),
+          matching: find.byType(BackdropFilter),
+        ),
+        findsNothing,
+      );
+
+      // Golden should display color tiles without CupertinoPopupSurface being
+      // displayed.
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.color-filter.removed.png'),
+      );
+    });
+  });
+  group('Surface Color', () {
```

Comment:
A long time ago I was told that `group` is discouraged due to having some subtle problems. Although I don't think they apply to your tests specifically, I think the convention in general is to avoid using `group`s, and just putting related tests together is enough.

Reply:
Removed all groups

Reply:
I don't know... I can't speak of any concrete cases to convincingly ban it, and I believe `group`s are still used somewhat throughout the repo. I just in general avoided it because I was told and it was not hard to avoid it either. :/ Maybe we can ask in the flutter-hackers chat to see how people feel about it. And feel free to keep your groups too.

Reply:
And also I'm still seeing groups all around? Did you forget to push your changes?